### PR TITLE
Added support for Cellular PPP interface

### PR DESCRIPTION
**Pull request by mikaleppanen**
When configured to use cellular interface, creates PPP and onboard
network stack instances, and connects cellular interface.

When PPP interface goes up, reads interface identifier from PPP
service and creates link local address based on it.

### CODE REVIEW

File: configs/Wisun_Stm_s2lp_RF_lte.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        "SERIAL_CTS": "NC",
+        "SERIAL_RTS": "NC",
+        "uc-channel-function": 255,
+        "bc-channel-function": 255,
+        "regulatory-domain": 3,
+        "operating-class": 255,
+        "operating-mode": 255,
+        "uc-fixed-channel": "0xffff",
+        "bc-fixed-channel": "0xffff",
+        "network-name": "\"wisuntest\"",
```
Comment (artokin) [inline configs/Wisun_Stm_s2lp_RF_lte.json]: Network name can be harmonized later.

**Comment by mikaleppanen** [comment]
Related mbed-os pull request: https://github.com/ARMmbed/mbed-os/pull/10974

File: configs/Wisun_Stm_s2lp_RF_lte.json (hunk 1 above)
Comment (mikaleppanen) [reply configs/Wisun_Stm_s2lp_RF_lte.json]: Corrected this to default name

**Comment by felipeLeast** [comment]
Hi @mikaleppanen and @artokin, I was tring to test this pull request and checked that there is no NanostackPPPInterface.h and PPPInterface.h files on mbed-os. Those files are in a privete repo? Or in a differente branch? If so, can I get a copy of them, I'm trying to do the same implementation but on Ublox C027 demo board.

Thank you,

**Comment by mikaleppanen** [comment]
Related mbed-os pull request: https://github.com/ARMmbed/mbed-os/pull/10974

**Comment by felipeLeast** [comment]
@mikaleppanen @artokin Which shield board or demo board are you using to test EC2x, are you using with K64F? Thank you,

**Comment by mikaleppanen** [comment]
@felipeLeast I have been using mainly K66F for testing.

**Comment by felipeLeast** [comment]
> @felipeLeast I have been using mainly K66F for testing.

I see, and how are you interfacing EC2x from Quectel?

**Comment by mikaleppanen** [comment]
@artokin @mikter Please review.

**Pull request merged**